src/engines/misc/master/audio.cpp

```diff
@@ -0,0 +1,34 @@
+#include "audio.hpp"
+
+#include "services/audio_manager.hpp"
+
+namespace otto::engines::master {
+
+  Audio::Audio() noexcept
+  {
+    
+  }
+
+  void Audio::action(itc::prop_change<&Props::volume>, float v) noexcept
+  {
+    volume_square_ = v * v;
+  }
+
+  void Audio::action(itc::prop_change<&Props::tempo>, int t) noexcept
+  {
+    tempo_ = t;
+  }
+
+  audio::ProcessData<2> Audio::process(audio::ProcessData<2> data) noexcept
+  {
+    for (auto&& l : data.audio[0]) {
+      l = util::math::fastatan( l * volume_square_ );
+    }
+    for (auto&& r : data.audio[1]) {
+      r = util::math::fastatan( r * volume_square_ );
```

Comment:
eventually, i think we want some real compression or something here...

Reply:
I agree. But I want to focus on it for a while, which is not really the best time right now...